src/Utilities/Compiler/CodeMetrics/MetricsHelper.cs

```diff
@@ -265,21 +265,30 @@ static bool hasConditionalLogic(IOperation operation)
         private static void AddCoupledNamedTypesCore(ImmutableHashSet<INamedTypeSymbol>.Builder builder, ITypeSymbol typeOpt)
         {
             if (typeOpt is INamedTypeSymbol usedType &&
-                !isIgnoreableType(usedType) &&
-                builder.Add(usedType))
```

Comment:
You can change this to `builder.Add(usedType.OriginalDefinition)` and will likely not need any other changes in this method.

Reply:
I changed it to `builder.Add(usedType.OriginalDefinition)` and reverted most of the code back to the way it was. We still need to take the call out of the `if` expression, though, to ensure that different generic arguments on the same generic types (`G<C1>` and `G<C2>`) cause their generic arguments' types (`C1` and `C2`) to be included.

Reply:
Can we still restore the bail out condition where build.Add returned false for a non-generic type?

Reply:
Ah, I see the only code within the if statement runs on generic types, so we should be fine.